handlers/ecs/summarization_v3/app.py

```diff
@@ -0,0 +1,327 @@
+import os
+import logging
+from enum import Enum
+from typing import Optional
+from datetime import date
+
+import requests
+import boto3
+import sentry_sdk
+
+from botocore.client import Config
+
+from langchain import OpenAI, PromptTemplate
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain.docstore.document import Document
+
+from langchain.chains.summarize import load_summarize_chain
+
+from fastapi import FastAPI, BackgroundTasks
+from pydantic import BaseModel
+
+from nlp_modules_utils import (
+    Database,
+    StateHandler,
+    prepare_sql_statement_success,
+    prepare_sql_statement_failure,
+    status_update_db,
+    upload_to_s3,
+    send_request_on_callback,
+    update_db_table_callback_retry
+)
+
+logging.getLogger().setLevel(logging.INFO)
+
+logging.info("Loaded module as ecs task.")
+
+SENTRY_DSN = os.environ.get("SENTRY_DSN")
+OPENAI_API_KEY = os.environ.get("OPENAI_API_KEY")
+ENVIRONMENT = os.environ.get("ENVIRONMENT")
+sentry_sdk.init(SENTRY_DSN, environment=ENVIRONMENT, attach_stacktrace=True, traces_sample_rate=1.0)
+
+# Note: boto3 initialization outside class to make it thread safe.
+# s3_client for generating presigned Url.
+s3_client = boto3.client(
+    "s3",
+    region_name=os.environ.get("AWS_REGION", "us-east-1"),
+    config=Config(
+        signature_version="s3v4",
+        s3={"addressing_style": "path"}
+    )
+)
+
+class InputStructure(BaseModel):
+    """Input Str"""
+    entries_url: Optional[str] = None
+    client_id: Optional[str] = None
+    callback_url: Optional[str] = None
+    summarization_id: Optional[str] = None
+
+ecs_app = FastAPI()
+
+@ecs_app.get("/")
+def home():
+    """ Test endpoint """
+    return "Welcome to the ECS Task of Summarization Module v3."
+
+@ecs_app.post("/generate_report")
+async def gen_report(item: InputStructure, background_tasks: BackgroundTasks):
+    """Generate reports"""
+    entries_url = item.entries_url
+    client_id = item.client_id
+    callback_url = item.callback_url
+    summarization_id = item.summarization_id
+
+    entries = reports_generator_handler.download_prepare_entries(entries_url=entries_url)
```

Comment:
reports_generator_handler is confusing here. Maybe define reports generator above this, or in a separate module?

Reply:
This is fixed.